### PR TITLE
Try split dbms (formats, core)

### PR DESCRIPTION
**Pull request by proller**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

File: dbms/src/Misc/ExternalTable.cpp [hunk 1]
```diff
@@ -1,16 +1,17 @@
+#include "ExternalTable.h"
+
 #include <boost/program_options.hpp>
 #include <boost/algorithm/string.hpp>
 #include <DataStreams/AsynchronousBlockInputStream.h>
 #include <DataTypes/DataTypeFactory.h>
 #include <Interpreters/Context.h>
 #include <IO/copyData.h>
-#include <IO/ReadBufferFromIStream.h>
+//#include <IO/ReadBufferFromIStream.h>
```
Comment (alesapin) [inline dbms/src/Misc/ExternalTable.cpp]: This lines should be removed?

File: dbms/src/Misc/ExternalTable.cpp [hunk 2]
```diff
@@ -1,16 +1,17 @@
+#include "ExternalTable.h"
```
Comment (alesapin) [inline dbms/src/Misc/ExternalTable.cpp]: Why not absolute path %<Misc/ExternalTable.h>%

File: dbms/src/Core/materializeBlock.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#include <DataStreams/materializeBlock.h>
+#include "materializeBlock.h"
```
Comment (alesapin) [inline dbms/src/Core/materializeBlock.cpp]: Let's rename header to `MaterializedBlock.h`.

File: dbms/src/Core/BackgroundSchedulePool.h [hunk 4]
```diff
@@ -1,156 +1 @@
...
-    /// Thread group used for profiling purposes
-    ThreadGroupStatusPtr thread_group;
-
-    void attachToThreadGroup();
-};
-
-using BackgroundSchedulePoolPtr = std::shared_ptr<BackgroundSchedulePool>;
-
-}
+#include <Misc/BackgroundSchedulePool.h>
```
Comment (alesapin) [inline dbms/src/Core/BackgroundSchedulePool.h]: Very dirty hack. Why we cannot use %<Misc/BackgroundSchedulePool.h>% everywhere?

File: dbms/src/Misc/NamesAndTypes.cpp [hunk 5]
```diff
@@ -0,0 +1,42 @@
+#include <Core/NamesAndTypes.h>
```
Comment (alesapin) [inline dbms/src/Misc/NamesAndTypes.cpp]: Header in Code and implementation in Misc? Is it legal?

**Comment by alesapin** [comment]
Maybe linkage became clearer, but code become messy. I don't think it's a good practice.

File: dbms/src/Core/materializeBlock.cpp (hunk 3 above)
Comment (proller) [reply dbms/src/Core/materializeBlock.cpp]: Header name should match function name.
M* correct for class

File: dbms/src/Core/BackgroundSchedulePool.h (hunk 4 above)
Comment (proller) [reply dbms/src/Core/BackgroundSchedulePool.h]: Its temporary hack, i will make mass replace to Misc/BackgroundSchedulePool.h if anything other is ok

File: dbms/src/Misc/NamesAndTypes.cpp (hunk 5 above)
Comment (proller) [reply dbms/src/Misc/NamesAndTypes.cpp]: Only small part of implementation in Misc (to not use DataTypes/DataTypeFactory.h in Core )

**Pull request closed without merging**